### PR TITLE
Cleans up mining turf code

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
Thanks @ChangelingRain you're a genius

### CODE REVIEW

**Comment by Cheridan** [comment]
-100 lines

:angel:

**Comment by xxalpha** [comment]
smooth

File: code/modules/mining/mine_turfs.dm [hunk 1]
```diff
@@ -274,7 +221,9 @@ var/global/list/rockTurfEdgeCache
 
 /turf/simulated/mineral/gibtonite/proc/defuse()
 	if(stage == 1)
-		icon_state = "rock_Gibtonite_inactive"
+		overlays -= activated_image
+		var/image/I = image('icons/turf/smoothrocks.dmi', loc = src, icon_state = "rock_Gibtonite_inactive", layer = 18)
```
Comment (ChangelingRain) [inline code/modules/mining/mine_turfs.dm]: layer 4.06

File: code/modules/mining/mine_turfs.dm [hunk 2]
```diff
@@ -239,7 +184,9 @@ var/global/list/rockTurfEdgeCache
 
 /turf/simulated/mineral/gibtonite/proc/explosive_reaction(mob/user = null, triggered_by_explosion = 0)
 	if(stage == 0)
-		icon_state = "rock_Gibtonite_active"
+		var/image/I = image('icons/turf/smoothrocks.dmi', loc = src, icon_state = "rock_Gibtonite_active", layer = 18)
```
Comment (ChangelingRain) [inline code/modules/mining/mine_turfs.dm]: see above

File: code/modules/mining/mine_turfs.dm [hunk 3]
```diff
@@ -1,20 +1,18 @@
...
+	canSmoothWith = list (/turf/simulated/mineral, /turf/simulated/wall)
 	baseturf = /turf/simulated/floor/plating/asteroid/airless
 	oxygen = 0
 	nitrogen = 0
 	opacity = 1
 	density = 1
+	pixel_y = -4
+	pixel_x = -4
 	blocks_air = 1
 	layer = TURF_LAYER + 0.05
```
Comment (ChangelingRain) [inline code/modules/mining/mine_turfs.dm]: change this to MOB_LAYER + 0.5

File: code/modules/mining/mine_turfs.dm [hunk 4]
```diff
@@ -49,50 +46,6 @@ var/global/list/rockTurfEdgeCache
...
-			T = get_step(src, WEST)
-			if (T)
-				T.overlays += rockTurfEdgeCache[EAST_EDGING]
-
-	if (mineralType && mineralAmt && spread && spreadChance)
-		for(var/dir in cardinal)
-			if(prob(spreadChance))
-				var/turf/T = get_step(src, dir)
-				if(istype(T, /turf/simulated/mineral/random))
-					Spread(T)
```
Comment (ChangelingRain) [inline code/modules/mining/mine_turfs.dm]: Kor you actually need 80-85

File: code/modules/mining/mine_turfs.dm [hunk 5]
```diff
@@ -49,50 +46,6 @@ var/global/list/rockTurfEdgeCache
...
-
-	HideRock()
-	return
-
-/turf/simulated/mineral/proc/HideRock()
-	if(hidden)
-		name = "rock"
-		icon_state = "rock"
```
Comment (ChangelingRain) [inline code/modules/mining/mine_turfs.dm]: This means all minerals will retain the name they have by default, did you mean to do that?

File: code/modules/mining/mine_turfs.dm [hunk 6]
```diff
@@ -147,18 +100,16 @@ var/global/list/rockTurfEdgeCache
 
 /turf/simulated/mineral/iron
 	name = "iron deposit"
-	icon_state = "rock_Iron"
 	mineralType = /obj/item/weapon/ore/iron
 	spreadChance = 20
 	spread = 1
-	hidden = 0
+	scan_state = "rock_Iron"
```
Comment (ChangelingRain) [inline code/modules/mining/mine_turfs.dm]: >hiding iron

advanced mining scanner as default item when

Comment (KorPhaeron) [inline code/modules/mining/mine_turfs.dm]: Nobody even wants or needs iron anyway

Comment (ChangelingRain) [inline code/modules/mining/mine_turfs.dm]: admittedly true

**Comment by xxalpha** [comment]
Shouldn't you be giving these turfs an underlay?

File: code/modules/mining/mine_turfs.dm (hunk 3 above)
Comment (KorPhaeron) [inline code/modules/mining/mine_turfs.dm]: I don't think it'd be good to have people end up underneath the walls if they hand tele into them

File: code/modules/mining/mine_turfs.dm (hunk 5 above)
Comment (KorPhaeron) [inline code/modules/mining/mine_turfs.dm]: I fixed this by removing the names

File: code/modules/mining/mine_turfs.dm (hunk 3 above)
Comment (ChangelingRain) [inline code/modules/mining/mine_turfs.dm]: but otherwise the edging won't go over people and that'd look bad!

Comment (ChangelingRain) [inline code/modules/mining/mine_turfs.dm]: also teleporting into rock really shouldn't happen tbh but that's not what the pr is about

File: code/modules/mining/mine_turfs.dm (hunk 4 above)
Comment (KorPhaeron) [inline code/modules/mining/mine_turfs.dm]: Readded this

File: code/modules/mining/mine_turfs.dm (hunk 3 above)
Comment (KorPhaeron) [inline code/modules/mining/mine_turfs.dm]: Yeah, if we ever stop the teleport proc from throwing people into dense turfs, we can make it a higher layer.

File: code/modules/mining/mine_turfs.dm (hunk 6 above)
Comment (newfren) [inline code/modules/mining/mine_turfs.dm]: I've seen drones often go looking for iron, and gulag prisoners who don't know how to mine sand tend to beeline to iron as well.

**Pull request merged**